prov/efa/src/efa_tp_def.h

```diff
@@ -21,30 +23,20 @@
 	lttng_ust_field_integer_hex(size_t, rxr_op_entry, rxr_op_entry) \
 	lttng_ust_field_integer_hex(size_t, context, context)
 
-LTTNG_UST_TRACEPOINT_EVENT(
-	EFA_TP_PROV,
-	post_send,
-	LTTNG_UST_TP_ARGS(
-		X_PKT_ARGS
-	),
-	LTTNG_UST_TP_FIELDS(
-		X_PKT_FIELDS  
-	)
-)
-LTTNG_UST_TRACEPOINT_LOGLEVEL(EFA_TP_PROV, post_send, LTTNG_UST_TRACEPOINT_LOGLEVEL_INFO)
-
-LTTNG_UST_TRACEPOINT_EVENT(
-	EFA_TP_PROV,
+LTTNG_UST_TRACEPOINT_EVENT_CLASS(EFA_TP_PROV, post_wr_id,
```

Comment:
post_send and post_recv are two different events. 

Both events have wr_id as an argument.

Reply:
Is your comment re: the tracepoints being otherwise identical? If so, the purpose would be to isolate each event if you wish. E.g.

```
lttng enable-event --userspace efa:post_recv
```

which would only execute the `efa_tracepoint_wr_id_post_recv()` traces.

Keep in mind these tracepoints existed before this PR.

Reply:
I was not familiar with `LTTNG_UST_TRACEPOINT_EVENT_CLASS` and `LTTNG_UST_TRACEPOINT_EVENT_INSTANCE`. 

I thought you changed the tracepoint's name from `post_send` and `post_recv` to `post_wr_id`.

While in fact, `post_wr_id` is the template you defined.

I think the name is a little confusing.

Reply:
Correct. On a side note, I spent some time trying to create some macros to clean it up a little bit - e.g. `EFA_TRACEPOINT_EVENT()` but something with the overall setup causes some preprocessor expansion issues with the nested macros' commas.

Do you have a preferred alternative to `post_wr_id` for the tracepoint class?

Reply:
I do not have better idea. I think it is good for now.